### DISCUSSION TITLE
What is wrong with my label_map_path?

### DISCUSSION THREAD

**Discussion by helmut72** [fluent/fluent-bit#7750] [Q&A]
This json:

```json
{
  "key1": "value1",
  "key2": {
    "key3": "value3",
    "key4": {
      "key5": "value5"
    }
  },
  "key6": "value6"
}
```

... works fine with this label_map_path:

```json
{
  "key1": "key1",
  "key2": {
    "key3": "key3",
    "key4": {
      "key5": "key5"
    }
  },
  "key6": "key6"
}
```

This json with square brackets:

```json
{
  "key1": "value1",
  "key2": {
    "key3": "value3",
    "key4": {
      "key5": [
        "value5"
      ]
    }
  },
  "key6": "value6"
}
```

... doesn't work with this label_map_path:

```json
{
  "key1": "key1",
  "key2": {
    "key3": "key3",
    "key4": {
      "key5": ["key5"]
    }
  },
  "key6": "key6"
}
```

Also not with this label_map_path:

```json
{
  "key1": "key1",
  "key2": {
    "key3": "key3",
    "key4": {
      "key5": "key5"
    }
  },
  "key6": "key6"
}
```

This label_map_path shows at least the whole content of `key4`:
(I have more square brackets in this tree than `key5`)

```json
{
  "key1": "key1",
  "key2": {
    "key3": "key3",
    "key4": "key4"
  },
  "key6": "key6"
}
```

Are square brakets supported and if yes, what is the syntax?

Thank you.

**Comment by patrick-stephens** [comment]
What do you mean by `local_map_path`? I can't find any reference to it in Fluent Bit docs or code: https://github.com/search?q=repo%3Afluent%2Ffluent-bit%20local_map_path&type=code

Do you possibly mean record accessor syntax? https://docs.fluentbit.io/manual/administration/configuring-fluent-bit/classic-mode/record-accessor

If not, do you have a specific issue with parsing or otherwise handling that JSON with Fluent Bit? If so, what is it, what version of Fluent Bit and config are you using, etc.? There's an issue template to capture it all.

**Reply by helmut72** [reply]
Oh sorry, replace local with label. I mean the `label_map_path` property in the Loki output of fluent-bit:
https://docs.fluentbit.io/manual/pipeline/outputs/loki#using-the-label_map_path-property

I've replaced local with label in my question. I use the docker-image of fluent-bit, 2.1.7. But see that today a new version was released. Changelog doesn't mention anything in the Loki output, but will try it later of course.

**Reply by helmut72** [reply]
Same with v2.1.8. Did I configure something wrong, is it a bug or a feature request, because square brackets are not supported for `label_map_path`?

**Reply by patrick-stephens** [reply]
I'm afraid I don't know, was just confused initially so wanted to clarify. You could ask on Slack to see if anyone has used it.

**Reply by helmut72** [reply]
I don't have a Slack user.
